### PR TITLE
tilt: switch to public opentracer endpoint, allow tracing from synclet

### PR DESCRIPTION
**Pull request by landism**
a lot of the diff is reorganizing the startup code so that:
1. we've parsed the args before we initialize the tracer (to allow --trace)
2. the same function is responsible for initializing and cleaning up the tracer (mostly motivated by broken ^C: when we canceled the context, we'd close the trace collector and back up out of the stack, and calls to `span.Finish` would hang)

Sample trace here: [here](https://ui.honeycomb.io/windmill/datasets/traces/result/8NSQj2RMLn3/trace/2e6d2cf534fecadc)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -31,9 +31,6 @@ benchmark:
 errcheck:
 	errcheck -ignoretests -ignoregenerated ./...
 
-start_tracer:
```
Comment (maiamcc) [inline Makefile]: is there no longer any setup code for the tracer? don't I need something running locally? are there setup steps if we're running this for the first time in a new cluster?

File: internal/tracer/tracer.go [hunk 2]
```diff
@@ -1,30 +1,50 @@
...
+}
+
+func (zl zipkinLogger) Log(keyvals ...interface{}) error {
+	logger.Get(zl.ctx).Debugf("%v", keyvals)
+	return nil
+}
+
+var _ zipkin.Logger = zipkinLogger{}
+
+func Init(ctx context.Context) (func() error, error) {
```
Comment (maiamcc) [inline internal/tracer/tracer.go]: this is dark sorcery, thanks for doing this!

File: Makefile (hunk 1 above)
Comment (landism) [reply Makefile]: No setup - it all goes to the globally public opentracing.windmill.build, regardless of running locally or in whatever cluster.

I've pushed its k8s config to https://github.com/windmilleng/honeycomb-opentracing-proxy, which might not be the best place for it.

Comment (landism) [reply Makefile]: I should say: I wanted to keep it opt-in, so users aren't unwittingly sending us their data, so `--trace` is now required if you want a trace.

**Pull request merged**